registry/handlers/manifests.go

```diff
@@ -65,17 +66,58 @@ type manifestHandler struct {
 
 // GetManifest fetches the image manifest from the storage backend, if it exists.
 func (imh *manifestHandler) GetManifest(w http.ResponseWriter, r *http.Request) {
-	ctxu.GetLogger(imh).Debug("GetImageManifest")
+	ctxu.GetLogger(imh).Debug("GetManifest")
 	manifests, err := imh.Repository.Manifests(imh)
 	if err != nil {
 		imh.Errors = append(imh.Errors, err)
 		return
 	}
+	supportsSchema2 := false
+	supportsManifestList := false
+	supportsOCISchema := false
+	supportsOCIManifestList := false
+	// this parsing of Accept headers is not quite as full-featured as godoc.org's parser, but we don't care about "q=" values
+	// https://github.com/golang/gddo/blob/e91d4165076d7474d20abda83f92d15c7ebc3e81/httputil/header/header.go#L165-L202
+	for _, acceptHeader := range r.Header["Accept"] {
+		// r.Header[...] is a slice in case the request contains the same header more than once
+		// if the header isn't set, we'll get the zero value, which "range" will handle gracefully
+
+		// we need to split each header value on "," to get the full list of "Accept" values (per RFC 2616)
+		// https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.1
+		for _, mediaType := range strings.Split(acceptHeader, ",") {
+			// remove "; q=..." if present
+			if i := strings.Index(mediaType, ";"); i >= 0 {
+				mediaType = mediaType[:i]
+			}
+
+			// it's common (but not required) for Accept values to be space separated ("a/b, c/d, e/f")
+			mediaType = strings.TrimSpace(mediaType)
+
+			if mediaType == schema2.MediaTypeManifest {
+				supportsSchema2 = true
+			}
+			if mediaType == manifestlist.MediaTypeManifestList {
+				supportsManifestList = true
+			}
+			if mediaType == ocischema.MediaTypeManifest {
+				supportsOCISchema = true
+			}
+			if mediaType == manifestlist.MediaTypeOCIManifestList {
+				supportsOCIManifestList = true
+			}
+		}
+	}
+	supportsOCI := supportsOCISchema || supportsOCIManifestList
 
 	var manifest distribution.Manifest
 	if imh.Tag != "" {
 		tags := imh.Repository.Tags(imh)
-		desc, err := tags.Get(imh, imh.Tag)
+		var desc distribution.Descriptor
+		if !supportsOCI {
+			desc, err = tags.Get(imh, imh.Tag)
+		} else {
+			desc, err = tags.Get(imh, imh.annotatedTag(false))
```

Comment:
The code here (which is actually used), doesn't work and just always retrieves the tag as literally passed in - it looks like there is a more evolved version in GetImageManifest.